main.rb

```diff
@@ -0,0 +1,158 @@
+    puts '7 - Exit'
+    user_option = gets.chomp
+    get_user_option(user_option)
+  end
+
+  def get_user_option(option)
+    case option
+    when '1'
+      all_books
+    when '2'
+      all_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      rentals_by_id
+    when '7'
+      abort('Thank you for using our Library!')
+    else
+      puts 'Please enter a number between 1 and 7'
+      puts
+      menu
+    end
+  end
+
+  def all_books
+    puts 'There are no books! Add a new book.' if @books.empty?
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    sleep 1
+    puts
+    menu
+  end
+
+  def all_people
+    puts 'The are no people! Please add a student or teacher.' if @people.empty?
+    @people.map { |person| puts "[#{person.class}] Name: #{person.name}, Id: #{person.id}, Age: #{person.age}, " }
+    sleep 1
+    puts
+    menu
+  end
+
+  def create_person
+    print 'Do you want to create a student(1) or a teacher(2)? [Input the number]:'
+    input = gets.chomp.to_i
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    case input
+    when 1
+      create_student(age, name)
+    when 2
+      create_teacher(age, name)
+    end
+    puts 'Person created successfully!'
+    menu
+  end
+
+  def create_book
+    print 'Title '
+    title = gets.chomp
+
+    print 'Author '
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    @books << book
+
+    puts 'Book added successfully'
+    sleep 1
+    puts
+    menu
+  end
+
+  def create_student(age, name)
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.upcase == 'Y'
+    @people << Student.new(name, age, parent_permission)
+  end
+
+  def create_teacher(age, name)
+    print 'Specialization: '
+    specialization = gets.chomp
+    @people << Teacher.new(name, age, specialization)
+  end
+
+  def create_rental
+    puts 'Select a book from the folowing list by number'
+    @books.each_with_index { |book, index| puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}" }
+    book_index = gets.chomp.to_i
+    puts book_index
+    puts 'Select a person from the folowing list by number (not id)'
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_index = gets.chomp.to_i
+    puts person_index
+    print 'Date: '
+    date = gets.chomp
+    @rentals << Rental.new(date, @books[book_index], @people[person_index])
+    puts 'Rental created successfully'
+    sleep 1
+    puts
+    menu
+  end
+
+  def rentals_by_id
+    puts @rentals
+    print 'ID of person: '
+    id = gets.chomp.to_i
+    puts 'Rentals:'
+    @rentals.each do |rental|
+      puts "Date: #{rental.date}, Book \"#{rental.book.title}\" by #{rental.book.author}" if rental.person.id == id
```

Comment:
The previous reviewer probably could not get to this point in the app because of the other errors.
- [ ] Let's keep our app output neat. Kindly remove the printing of the rental in line 140 above. As shown in the first image below it makes your app look messy.
- [ ] I am unable to list the rental I created because of the error in the first image below. There is a mismatch of **`id`** definition. This mismatch occurred in how you define the Rental class and how you are calling it.

  In line 132 above, your format of calling the Rental class is `date`, `book`, and `person`; whereas your Rental class is initialized as `date`, `person` and `book` (as shown in the second image). Simply making the definition of the parameters in the Rental `initialize` method and line 132 to be uniform will resolve this.

**First image**
  ![yaseer-rent](https://user-images.githubusercontent.com/36542986/151658825-d805dceb-f637-4ae9-bc35-0dcbb9d13cd8.jpg)

**Second image**
  ![rent-init](https://user-images.githubusercontent.com/36542986/151659014-15728c8d-2141-4482-8974-e6269d0e3347.jpg)